src/contracts/TokenForwarding.cdc

```diff
@@ -26,32 +26,43 @@ pub contract TokenForwarding {
         // This is where the deposited tokens will be sent.
         // The type indicates that it is a reference to a receiver
         //
-        access(self) var recipient: &{FungibleToken.Receiver}
+        access(self) var recipient: Capability
 
         // deposit
         //
         // Function that takes a Vault object as an argument and forwards
         // it to the recipient's Vault using the stored reference
         //
         pub fun deposit(from: @FungibleToken.Vault) {
-            emit ForwardedDeposit(amount: from.balance, from: self.owner?.address)
-            self.recipient.deposit(from: <-from)
+            let receiverRef = self.recipient.borrow<&{FungibleToken.Receiver}>()!
+
+            let balance = from.balance
+          
+            receiverRef.deposit(from: <-from)
+
+            emit ForwardedDeposit(amount: balance, from: self.owner?.address)
         }
 
         // changeRecipient changes the recipient of the forwarder to the provided recipient
         //
-        pub fun changeRecipient(_ newRecipient: &{FungibleToken.Receiver}) {
+        pub fun changeRecipient(_ newRecipient: Capability) {
+            pre {
+                newRecipient.borrow<&{FungibleToken.Receiver}>() != nil: "Could not borrow Receiver reference from the Capability"
+            }
```

Comment:
What's the reason for adding this? Ensuring the capability can be borrowed in this setter doesn't really ensure it can be borrowed later. Same below

Reply:
Its not perfect, but even though it can be changed later potentially, it is good to know that at least it works when you initially set the value

Reply:
Does that make sense @turbolent? Are you ok with me merging it?

Reply:
Yep, that makes sense